include/urg_node/urg_c_wrapper.hpp

```diff
@@ -102,6 +102,9 @@ struct SerialConnection
   int serial_baud;
 };
 
+static const size_t AR00_PACKET_SIZE = 4379;
+static const size_t DL00_PACKET_SIZE = 1936;
```

Comment:
Are these actually the same for all Hokuyo lasers? That seems unlikely to me, since different lasers have different number of readings per scan

Reply:
My understanding is these commands (AR00 and DL00) are separate from the SCIP2.0 protocol which the urg_library is based on and are exclusively part of the UAM protocol specification (for safety LIDARs of which the UAM-05LP is the only one). I've attached the protocol spec.  

[UAM05LP_CommunicationSpecification_en_2_4_0_r1.pdf](https://github.com/ros-drivers/urg_node/files/11065525/UAM05LP_CommunicationSpecification_en_2_4_0_r1.pdf)

Hence my confidence in setting these as fixed values as anyone using these detailed status commands would need to be doing it with a UAM-05LP scanner as non-safety scanners wouldn't support the UAM commands. It's a bit of a hack the way it's been implemented at the moment effectively switching protocols mid stream to get the safety data and then going back to SCIP to get the scan data again. 

Ideally for safety scanners you'd switch to using the UAM protocol exclusively but that's a much bigger project as it would be good to integrate it into the urg_c library.

Reply:
Happy to leave it open for people to chime in with other experiences.

As a middle ground we can also set the limit to the maximum string index AR00 and DL00 update functions.

Reply:
That seems reasonable - I'm traveling for a few days - but I will test this out on the couple different Hokuyos I have when I get home next week to make sure it works as expected on the lower end lasers

Reply:
I've got a UST-10LX at home and ran a quick check, as expected with detailed_status = true the command fails and returns an empty packet. You do get some nasty socket errors though and the sensor is unhappy. 

I just pushed a quick product name get to the detailed_status branch of the getStatus() function which gives a nice friendly warning if the command isn't supported. 

Let me know what you think?